packages/protocol/contracts/L1/verifiers/SgxVerifier.sol

```diff
@@ -77,37 +81,29 @@ contract SgxVerifier is EssentialContract, IVerifier {
         ids = _addInstances(_instances);
     }
 
-    /// @notice Adds SGX instances to the registry by another SGX instance.
-    /// @param id The id of the SGX instance who is adding new members.
-    /// @param newInstance The new address of this instance.
-    /// @param extraInstances The address array of SGX instances.
-    /// @param signature The signature proving authenticity.
-    /// @return ids The respective instanceId array per addresses.
-    function addInstances(
-        uint256 id,
-        address newInstance,
-        address[] calldata extraInstances,
-        bytes calldata signature
-    )
+    /// @notice Adds an SGX instance after the attestation is verified
+    /// @param attestation The parsed attestation quote.
+    /// @return id The respective instanceId
+    function registerInstance(V3Struct.ParsedV3QuoteStruct calldata attestation)
```

Comment:
Security feature idea: use the already stored `addedAt` so that SGX instances that are registered with onchain RA verification are only usable after some short time delay. This we can monitor for issues in this smart contract and reject instances that actually don't have a valid RA.